scripts/background.js

```diff
@@ -145,85 +157,114 @@ function processHeaders(details) {
+                redeemOccurred = attemptRedeem(url, details.tabId);
+            } else if (countStoredTokens() == 0) {
+                // Update icon to show user that token may be spent here
+                updateIcon("!");
             }
-            // Update icon to show user that token may be spent here
-            updateIcon("!");
+        }
+
+        // If signing is permitted then we should note this
+        if (!redeemOccurred && DO_SIGN) {
+            readySign = true;
         }
     }
 }
 
-// Attempts to redeem a token if we should do
+// Attempts to redeem a token
+// Returns true if a redemption is fired, and false otherwise.
 function attemptRedeem(url, respTabId) {
     // Prevent reloading on captcha.website
     if (url.host.indexOf(CHL_CAPTCHA_DOMAIN) != -1) {
-        return;
+        return false;
     }
 
     // Check all cookie stores to see if a clearance cookie is held
-    chrome.cookies.getAllCookieStores(function(stores) {
-        let clearanceHeld = false;
-        stores.forEach( function(store, index) {
-            var tabIds = store.tabIds;
-            if (tabIds.length && tabIds[0].id !== undefined) {
-                tabIds = tabIds.map((tab) => respTabId.id);
-            }
-            var storeMatches = tabIds.indexOf(respTabId) >= 0;
-            if (storeMatches) {
-                chrome.cookies.get({"url": url.href, "name": CHL_CLEARANCE_COOKIE, "storeId": store.id}, function(cookie) {
-                    // Require an existing, non-expired cookie.
-                    if (cookie) {
-                        clearanceHeld = (cookie.expirationDate * 1000 >= Date.now());
-                    }
-                });
+    let fired = false;
+    if (CHECK_COOKIES) {
+        chrome.cookies.getAllCookieStores(function(stores) {
+            let clearanceHeld = false;
+            stores.forEach( function(store, index) {
+                var tabIds = store.tabIds;
+                if (tabIds.length && tabIds[0].id !== undefined) {
+                    tabIds = tabIds.map((tab) => respTabId.id);
+                }
+                var storeMatches = tabIds.indexOf(respTabId) > -1;
+                if (storeMatches) {
+                    chrome.cookies.get({"url": url.href, "name": CHL_CLEARANCE_COOKIE, "storeId": store.id}, function(cookie) {
+                        // Require an existing, non-expired cookie.
+                        if (cookie) {
+                            clearanceHeld = (cookie.expirationDate * 1000 >= Date.now());
+                        }
+                    });
+                }
+            });
+
+            // If a clearance cookie is not held then set the spend flag
+            if (!clearanceHeld) {
+                fireRedeem(url, respTabId);
+                fired = true;
             }
         });
+    } else {
+        // If cookies aren't checked then we always attempt to redeem.
+        fireRedeem(url, respTabId);
+        return true;
```

Comment:
```suggestion
        fired = true
```
I think changing this line would make it clearer that a redemption has been fired